File: tests/C/0018-acc+omp.c
int foo (int a, int b)
{
	#pragma acc parallel copyin(a) copyin(b)
	#pragma omp target teams map(to:a,b)
	{
	}

	#pragma acc parallel copy(a) copy(b)
	#pragma omp target teams \
	  map(a,b)
	{
	}

	#pragma acc parallel copy(a) copy(b)
	#pragma omp \
	  target \
	  teams \
	  map(a,b)
	{
	}

	int r = 0;
	#pragma acc parallel copy(a)
	#pragma acc loop reduction(+:r)
	#pragma omp target teams
	#pragma omp loop reduction(+:r)
	for (int i = a; i < b; ++i)
		r += i*i;

	return r;
}

File: tests/Fortran/Free/0018-acc+omp.f90
program test
integer :: a, b, r, i

!$acc serial copy(a) copy(b)
!$omp target map(a,b)
  b = 2*a
!$acc end serial
!$omp end target

!$acc serial copy(a) &
!$acc  copy(b)
!$omp target &
!$omp  map(a,b)
  b = 2*a
!$acc end serial
!$omp end target

!$acc serial copy(a) &
!$acc  copy(b)
!$omp target &
!$omp  map(tofrom:a) &
!$omp  map(tofrom:b)
  b = 2*a
!$acc end serial
!$omp end target

r = 0
!$acc parallel
!$acc loop reduction(+:r)
!$omp target teams
!$omp loop reduction(+:r)
do i = a, b
  r = r + a*a
end do
!$omp end target teams
!$acc end parallel

end program
